### PR TITLE
Add package requirements to helm chart

### PR DESCRIPTION
**Pull request by TerryHowe**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jonahjon — APPROVED** [review]

File: Common.mk [hunk 1]
```diff
@@ -456,10 +457,10 @@ validate-checksums: $(BINARY_TARGETS)
 helm/build: $(OUTPUT_DIR)/ATTRIBUTION.txt
 helm/build: $(if $(filter true,$(REPO_NO_CLONE)),,$(HELM_GIT_CHECKOUT_TARGET))
 helm/build: $(if $(wildcard $(MAKE_ROOT)/helm/patches),$(HELM_GIT_PATCH_TARGET),)
-	HELM_REGISTRY=$(IMAGE_REPO) \
 	IMAGE_TAG=$(IMAGE_TAG) \
+	HELM_IMAGE_LIST=$(HELM_IMAGE_LIST) \
```
Comment (jaxesn) [inline Common.mk:461]: Dont worry about changing this now, but something for the future.  This call is a bit out of hand considering its setting env vars and passing in a number of args.  No real suggestions, just a note.

**Review by jaxesn — APPROVED** [review]
Oh I like this!  

Looks good to me, ill let someone on your team merge, but ping me if you want me to go ahead.

File: build/lib/image_shasum.sh [hunk 2]
```diff
@@ -22,10 +22,13 @@ REGISTRY="${1?First argument is registry}"
 REPOSITORY="${2?Second argument is repository}"
 IMAGE_TAG="${3?Third argument is image tag}"
 
-JSONPATH="imageDetails[?(@.imageTags[?(@ == '"${IMAGE_TAG}"')])].imageDigest"
-if [[ "${REGISTRY}" == *"public.ecr.aws"* ]]
+if [ "${REGISTRY}" == "316434458148.dkr.ecr.us-west-2.amazonaws.com" ]
```
Comment (jonahjon) [inline build/lib/image_shasum.sh:25]: Should this be changed back?

Comment (TerryHowe) [reply build/lib/image_shasum.sh:25]: No, we are using skopeo now to get the shasum and this is for in the case of presubmits, we are not going to have a image to get a shasum, we'll just use "latest" so we can test building the helm chart

Comment (jonahjon) [reply build/lib/image_shasum.sh:25]: O so it's ok to include the account ID here then?

**Comment by TerryHowe** [comment]
Finally got it!

File: projects/fluxcd/flux2/Makefile [hunk 3]
```diff
@@ -25,12 +25,7 @@ HELM_SOURCE_OWNER=fluxcd-community
 HELM_SOURCE_REPOSITORY=helm-charts
 HELM_GIT_TAG?=$(shell cat HELM_GIT_TAG)
 HELM_DIRECTORY=charts/flux2
-HELM_ADDITIONAL_KEY_VALUES=\
-  FLUX_CLI_TAG=$(shell $(BASE_DIRECTORY)/build/lib/image_shasum.sh ${IMAGE_REPO} fluxcd/flux-cli $(LATEST)) \
-  HELM_CONTROLLER_TAG=$(shell $(BASE_DIRECTORY)/build/lib/image_shasum.sh ${IMAGE_REPO} fluxcd/helm-controller $(LATEST)) \
-  KUSTOMIZE_CONTROLLER_TAG=$(shell $(BASE_DIRECTORY)/build/lib/image_shasum.sh ${IMAGE_REPO} fluxcd/kustomize-controller $(LATEST)) \
-  NOTIFICATION_CONTROLLER_TAG=$(shell $(BASE_DIRECTORY)/build/lib/image_shasum.sh ${IMAGE_REPO} fluxcd/notification-controller $(LATEST)) \
-  SOURCE_CONTROLLER_TAG=$(shell $(BASE_DIRECTORY)/build/lib/image_shasum.sh ${IMAGE_REPO} fluxcd/source-controller $(LATEST))
+HELM_IMAGE_LIST="fluxcd/flux-cli fluxcd/helm-controller fluxcd/kustomize-controller fluxcd/notification-controller fluxcd/source-controller"
```
Comment (jaxesn) [inline projects/fluxcd/flux2/Makefile:28]: Oh i like this!

File: projects/goharbor/harbor/helm/sedfile.template [hunk 4]
```diff
@@ -1,15 +1,6 @@
...
-s/digest: {{ REGISTRYCTL_IMAGE_DIGEST }}$/digest: ${REGISTRYCTL_IMAGE_DIGEST}/
-s/digest: {{ CHARTMUSEUM_IMAGE_DIGEST }}$/digest: ${CHARTMUSEUM_IMAGE_DIGEST}/
-s/digest: {{ TRIVY_ADAPTER_IMAGE_DIGEST }}$/digest: ${TRIVY_ADAPTER_IMAGE_DIGEST}/
-s/digest: {{ NOTARY_SERVER_IMAGE_DIGEST }}$/digest: ${NOTARY_SERVER_IMAGE_DIGEST}/
-s/digest: {{ NOTARY_SIGNER_IMAGE_DIGEST }}$/digest: ${NOTARY_SIGNER_IMAGE_DIGEST}/
-s/digest: {{ DATABASE_IMAGE_DIGEST }}$/digest: ${DATABASE_IMAGE_DIGEST}/
-s/digest: {{ REDIS_IMAGE_DIGEST }}$/digest: ${REDIS_IMAGE_DIGEST}/
-s/digest: {{ EXPORTER_IMAGE_DIGEST }}$/digest: ${EXPORTER_IMAGE_DIGEST}/
\ No newline at end of file
+s,{{harbor/harbor-chartmuseum}},sha256:f5ac705ce6173725ac2f9da9891d13f1000e78caddbeca36e8c1c5b12e0f86fc,g
```
Comment (jaxesn) [inline projects/goharbor/harbor/helm/sedfile.template:3]: What are these shas?

**Review by jaxesn — APPROVED** [review]
small question, but this is an awesome improvement!

File: projects/goharbor/harbor/helm/sedfile.template (hunk 4 above)
Comment (TerryHowe) [reply projects/goharbor/harbor/helm/sedfile.template:3]: They are bogus shas for now. They won't work, but those images are not used.

**Review by jaxesn — APPROVED** [review]
/lgtm

**Comment by TerryHowe** [comment]
Rebased to see if that will fix that presubmit. Otherwise, I'm ignoring

**Review by jaxesn — APPROVED** [review]
/lgtm

**Pull request merged**